src/Sarif.UnitTests/Visitors/InsertOptionalDataVisitorTests.cs

```diff
@@ -0,0 +1,157 @@
+﻿// Copyright (c) Microsoft. All rights reserved. Licensed under the MIT        
+// license. See LICENSE file in the project root for full license information. 
+
+using System;
+using System.Globalization;
+using System.IO;
+using System.Text;
+using FluentAssertions;
+using Microsoft.CodeAnalysis.Sarif.Readers;
+using Microsoft.CodeAnalysis.Sarif.UnitTests;
+using Newtonsoft.Json;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.CodeAnalysis.Sarif.Visitors
+{
+    public class InsertOptionalDataVisitorTests : FileDiffingTests
+    {
+        // To rebaseline all test files set this value to true and rerun the testa
```

Comment:
>To rebaseline all test files set this value to true and rerun the testa [](start = 11, length = 71)

Neither the `Update-Expected.ps1` script in `Sarif.Multitool.FunctionalTests` nor `UpdateBaselines.ps` in `Sarif.FunctionalTests` require a special build to update the expected files. Can't this one be managed in the same way?

Reply:
Ok, I understand why. Both of those scripts update _all_ the expected files, and since they don't take care to make the files deterministic, they all change. We either need to
1) Make the expected files deterministic, or
2) Have an elaborate script that identifies the failing tests (say, by reading the .trx), identifies the files that need to be updated (say, by a naming convention, based on the name of the test), and updates only the necessary files.

---
In reply to: [205982194](https://github.com/Microsoft/sarif-sdk/pull/974#discussion_r205982194) [](ancestors = 205982194)